src/emulator/events/types.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * The types in this file are stolen from the firebase-functions SDK. They are not subject
+ * to change because they are TS interpretations of protobuf wire formats already used in
+ * productiton services.
+ */
+
+/**
+ * Wire formal for v1beta1 EventFlow.
+ * Used by Firestore and some other services.
+ */
+export interface LegacyEvent {
+  data: any;
+  eventType?: string;
+  resource?: string;
+  eventId?: string;
+  timestamp?: string;
+  params?: { [option: string]: any };
+  auth?: AuthMode;
+}
+
+/**
+ * Wire format for v1beta2 Eventflow (and likely v1).
+ * Used by PubSub, RTDB, and some other services.
+ */
+export interface Event {
```

Comment:
Same as above, let's dedupe if we can from [here](https://github.com/firebase/firebase-functions/blob/68a669183f61002c269781b3957bf9b74f10bd87/src/cloud-functions.ts#L47)